ras_backstage/controllers/secure_messaging_controller.py

```diff
@@ -101,3 +103,21 @@ def update_draft(encoded_jwt, message_json):
     message = json.loads(response.text)
     logger.debug('Draft updated successfully', message_id=message['msg_id'])
     return message
+
+
+def _create_authorization_header(encoded_jwt):
+    if current_app.config.get('USE_UAA'):
+        logger.info(f'Received token {encoded_jwt}')
+        sm_token = convert_token(encoded_jwt)
+    else:
+        sm_token = jwt.encode({'party_id': 'BRES', 'role': 'internal'},
+                              app.config['RAS_SECURE_MESSAGING_JWT_SECRET'],
+                              algorithm='HS256')
+    return {"Authorization": sm_token}
+
+
+def convert_token(access_token):
+    user_id = token_decoder.get_user_id(access_token)
+
+    secret = app.config['RAS_SECURE_MESSAGING_JWT_SECRET']
+    return jwt.encode({'party_id': user_id, 'role': 'internal'}, secret, algorithm='HS256')
```

Comment:
I think we could remove the convert_token method and remove some duplication to make this clearer, something like:
```
def _create_authorization_header(encoded_jwt):
    if current_app.config.get('USE_UAA'):
        logger.info(f'Received token {encoded_jwt}')
        party_id = token_decoder.get_user_id(access_token)
    else:
        party_id = 'BRES'	

	secret = app.config['RAS_SECURE_MESSAGING_JWT_SECRET']
	return jwt.encode({'party_id': party_id, 'role': 'internal'}, secret, algorithm='HS256')
```

Reply:
Agreed, will change